java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/mapping/SourceRecordMapperTest.java

```diff
@@ -0,0 +1,125 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.azure.eventhubs.mapping;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.from;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.lenses.streamreactor.connect.azure.eventhubs.source.TopicPartitionOffsetProvider.AzureOffsetMarker;
+import io.lenses.streamreactor.connect.azure.eventhubs.source.TopicPartitionOffsetProvider.AzureTopicPartitionKey;
+import java.util.Iterator;
+import java.util.Optional;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.jupiter.api.Test;
+
+class SourceRecordMapperTest {
+  private static final String TOPIC = "topic";
+  private static final Integer PARTITION = 10;
+  private static final Long OFFSET = 111L;
+  private static final Long TIMESTAMP = 2024L;
+  private static final String HEADER_KEY = "headerKey";
+  private static final String OUTPUT_TOPIC = "OUTPUT";
+
+  @Test
+  void shouldMapSourceRecordIncludingHeaders() {
+    //given
+    AzureTopicPartitionKey topicPartitionKey = new AzureTopicPartitionKey(TOPIC, PARTITION);
+    AzureOffsetMarker azureOffsetMarker = new AzureOffsetMarker(OFFSET);
+
+    byte[] exampleHeaderValue = new byte[] {1, 10};
+    int headerLength = exampleHeaderValue.length;
+    Header mockedHeader = mock(Header.class);
+    when(mockedHeader.key()).thenReturn(HEADER_KEY);
+    when(mockedHeader.value()).thenReturn(exampleHeaderValue);
+
+    Iterator<Header> iterator = singletonList(mockedHeader).iterator();
+    Headers mockedHeaders = mock(Headers.class);
+    when(mockedHeaders.iterator()).thenReturn(iterator);
+
+    ConsumerRecord<String, String> consumerRecord = mockConsumerRecord(Optional.of(mockedHeaders));
+
+    //when
+    Schema stringSchema = Schema.STRING_SCHEMA;
+    Schema optionalStringSchema = Schema.OPTIONAL_STRING_SCHEMA;
+    SourceRecord sourceRecord = SourceRecordMapper.mapSourceRecordIncludingHeaders(
+        consumerRecord, topicPartitionKey, azureOffsetMarker,
+        OUTPUT_TOPIC, optionalStringSchema, stringSchema);
+
+    //then
+    assertRecordAttributesAreMappedFromSourceConsumerRecord(sourceRecord, consumerRecord,
+        OUTPUT_TOPIC, optionalStringSchema, stringSchema, topicPartitionKey, azureOffsetMarker);
+    verify(consumerRecord).headers();
+    assertThat(sourceRecord.headers()).isNotEmpty().hasSize(1);
```

Comment:
if it has size(1) isn't it implicitly not empty?  Not sure you need both conditions.

Reply:
Streamlined it a bit.